subprojects/docs/src/docs/userguide/troubleshooting/validation_problems.adoc

```diff
@@ -264,3 +264,63 @@ someTask {
     inputFile.from(producer.someFile)
 }
 ```
+
+[[input_does_not_exist]]
+== Input doesn't exist
+
+This error occurs whenever a file (or a directory) is declared as an input of a task, but at the moment the task is executed, the file (or directory) doesn't exist.
+
+Usually, this hints at a missing task dependency: the file should exist _before_ the task is executed, which means that a dependent task wasn't executed.
+
+The symptoms are similar to <<implicit_dependency>> except that in this case the task which creates the file hasn't been executed.
+
+Please refer to the <<implicit_dependency>> section for possible solutions.
+If the file isn't produced by another task, you may want to make sure that it exists before the task is called.
+
+[[unexpected_input_type]]
+== Unexpected input file or directory
+
+This error indicates that a property expected a regular file as an input but that it was provided with a directory (or the other way around).
+
+For example, if property is annotated with `@InputFile`:
+
+```groovy
+@InputFile
+File getInputFile()
+```
+
+Then Gradle expects the input file to be a _regular file_. If the input is a directory, then validation fails.
+
+To fix this problem, you have two options:
+
+- either you made a mistake an provided a directory instead of a file in which case you just need to fix the input
+- or the task should actually have used a directory as an input, in which case you need to change the type of the property to `@InputDirectory`
+
+[[cannot_write_output]]
+== Cannot write to an output file or directory
+
+This error indicates that :
+
+- an output directory cannot be written because the directory property which has been configured actually refers to a regular file (or something else than an actual directory).
+- or that an output file cannot be written because the file property which has been configured actually refers to a directory
+
+For example, you've set an output directory to `/some/path/file.txt` instead of `/some/path`.
+It's also possible that you have configured an output directory like `/some/path` but that an ancestor `/some` is a regular file.
+
+To fix this problem, make sure that the configured output is a directory (for properties which expect a directory) or a file (for tasks which expect a file).
+
+[[cannot_write_to_reserved_location]]
+== Cannot write to reserved location
+
+This error indicates that you are trying to write a file to a location which is managed by Gradle only.
+Typically, this happens whenever you're trying to write a file directly into an artifact transforms output directory.
+If you did this intentionally, this is a mistake because those directories should never be written directly: all artifact transform writes should be performed within the artifact transform code itself.
+
+If you didn't intend to write in this directory, you should simply setup your task to write in a different location.
+
+[[unsupported_notation]]
+== Unsupported notation in file collection
```

Comment:
I am not sure this is about unsupported notations in file collections. Isn't this more about things which can't be resolved to a single location? At least that is a problem for `@InputFile` and `@InputDirectory`, right?

Reply:
Yes you're right it's not limited to file collections.